File: s4s/public/js/custom_user.js
frappe.ui.form.on("User",{
    after_save:function(frm){
        console.log("running")
        frappe.call({
            method:"s4s.public.python.custom_user.fetch_data",
            callback:function(r){
                // frm.save()
                // frm.save()
            }
        })
    }
})

File: s4s/public/python/custom_user.py
import frappe
from frappe import _

@frappe.whitelist()
def fetch_data():
    print("running"*100)
    doc = frappe.get_all("User",filters={'enabled':1},fields=['name','mobile_no'])
    otp = frappe.get_doc("S4S OTP Verification")
    otp.users = []
    for i in doc:
        otp.append("users",{
            "user":i.name,
            "mobile_number":i.mobile_no
            # "timestamp":"0",
            # "otp":"0"
        })
    otp.save()
    otp.save()